packages/react/focus-scope/src/FocusScope.tsx

```diff
@@ -113,7 +121,7 @@ const FocusScope = React.forwardRef((props, forwardedRef) => {
       container.dispatchEvent(mountEvent);
       if (!mountEvent.defaultPrevented) {
         focusFirst(getTabbableCandidates(container), { select: true });
-        if (document.activeElement === PREVIOUSLY_FOCUSED_ELEMENT) {
+        if (document.activeElement === previouslyFocusedElement) {
```

Comment:
The meaning of this conditional has now changed I think because of the newly added var, we need to be careful about this.

Before this line was there to make sure we focused the container if the focus was still where it was before mounting (even after running all the successive potential `focusFirst`).

Now it means something else because of the added `unmountAutoFocusElement` that can be arbitrarily passed in.

I think we basically still need the concept of `previouslyFocusedElement` (being `document.activeElement` at the point of mount) separate from the new var.